frontend/src/components/contentDisplay/SlideDisplayFactory.vue

```diff
@@ -0,0 +1,132 @@
+<!-- Visualizes a slide -->
+<template>
+  <div
+    class="aspect-video p-5 rounded-md"
+    :style="{ backgroundColor: slide.backgroundColor }"
+  >
+    <div class="grid grid-layout h-full" ref="wrapper">
+      <ContenDisplayFactory
+        v-for="slot in requiredSlots"
+        :key="slot"
+        :content="slide.content[slot]"
+        :style="getSlotStyle(slot)"
+        :edit-mode="editMode"
+      />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import {
+  Layout,
+  LayoutSlot,
+  LayoutType,
+} from '../../../../model/slide/layout/Layout';
+import { ISlide } from '../../../../model/slide/ISlide';
+import { PropType, Ref, onMounted, provide, ref } from 'vue';
+import ContenDisplayFactory from './ContenDisplayFactory.vue';
+
+const props = defineProps({
+  /**
+   * The slide to display
+   */
+  slide: {
+    type: Object as PropType<ISlide>,
+    required: true,
+  },
+  /**
+   * Indicates whether student or teacher is viewing the slide
+   */
+  editMode: {
+    type: Boolean,
+    required: true,
+    default: false,
+  },
+});
+
+interface GridSlot {
+  gridRowStart: number;
+  gridRowEnd: number;
+  gridColumnStart: number;
+  gridColumnEnd: number;
+}
+
+const map: Record<LayoutSlot, GridSlot> = {
+  0: { gridRowStart: 1, gridRowEnd: 2, gridColumnStart: 1, gridColumnEnd: 3 },
+  1: { gridRowStart: 2, gridRowEnd: 4, gridColumnStart: 1, gridColumnEnd: 2 },
+  2: { gridRowStart: 2, gridRowEnd: 4, gridColumnStart: 2, gridColumnEnd: 3 },
+  3: { gridRowStart: 2, gridRowEnd: 4, gridColumnStart: 2, gridColumnEnd: 3 },
+  4: { gridRowStart: 2, gridRowEnd: 4, gridColumnStart: 2, gridColumnEnd: 3 },
+};
+
+/**
+ * Gets the drid style for a given slot in the slides layout.
+ *
+ * @param slot Slot to check for
+ * @returns Grid style for the slot
+ */
+function getSlotStyle(slot: LayoutSlot): GridSlot {
```

Comment:
Wouldnt a map like `Record<LayoutType,Partial<Record<LayoutSlot,GridSlot>>>` be easier to understand here?
So we'd have
```
{
  [LayoutType.TITLEPAGE]: {
    [LayoutSlot.HEADER]: { gridRowStart: 1, gridRowEnd: 2, gridColumnStart: 1, gridColumnEnd: 3 }
  },
  ... more layouts
}
```

So we could more easily read / change layoutSlots in individual layouts